pex/environment.py

```diff
@@ -368,6 +368,10 @@ def _activate(self):
         with TRACER.timed('Adding sitedir', V=2):
           site.addsitedir(dist.location)
 
-        self._declare_namespace_packages(dist)
+    # NB: we use this loop, rather than putting the below line in the prior loop, to ensure that
+    # 'sys.path' contains all of the resolved dists before calling
```

Comment:
This comment misses the real point - the issue has nothing to do with namespace packages per se for example as pointed out in #713. The issue is if any pex bootstrap code whatsoever imports `pkg_resources` or some sub-package prior to completeing sys.path modification then any pex user code or requirements that use the pkg_resources global working set will potentially break. This is whac-a-mole. I'll dig a bit this morning to see if there is a more robust solution.

Reply:
Not alot more robust, but I think a bit better containment and, hopefully, a bit clearer explanation of the case we're tiptoeing around here #732.